packages/compass-aggregations/src/modules/index.js

```diff
@@ -675,6 +676,19 @@ export const getPipelineFromIndexedDB = (pipelineId) => {
   };
 };
 
+
+/**
+ * Make view pipeline.
+ *
+ * @returns {Array} The mapped/filtered view pipeline.
+ */
+export const makeViewPipeline = (unfilteredPipeline) => {
+  return unfilteredPipeline
+    .map((p) => (p.executor || generateStage(p)))
+    // generateStage can return {} under various conditions
+    .filter((stage) => !isEmpty(stage));
```

Comment:
Maybe we can change `generateStage` in a way that returns something that is easier to identify as an empty stage (like `null` or something) and then we don't need this dependency on `isEmpty` to do the check (lodash is used a lot across the codebase, but we are trying to use it less recently and remove it where appropriate), wdyt?

Reply:
I considered that and had a look at the other places (outside of tests) that call generateStage():

* https://github.com/mongodb-js/compass/blob/fe3c919c6a5bb5f8967bdd3777f1e277268787fb/packages/compass-aggregations/src/utils/stage.js#L27
* https://github.com/mongodb-js/compass/blob/fe3c919c6a5bb5f8967bdd3777f1e277268787fb/packages/compass-aggregations/src/modules/update-view.js
* https://github.com/mongodb-js/compass/blob/fe3c919c6a5bb5f8967bdd3777f1e277268787fb/packages/compass-aggregations/src/modules/pipeline.js#L516
* https://github.com/mongodb-js/compass/blob/fe3c919c6a5bb5f8967bdd3777f1e277268787fb/packages/compass-aggregations/src/modules/pipeline.js#L565

We'd have to change those to turn `null` back into `{}` because that seems to be what they expect. So I'm not sure - it seems debatable either way.

Reply:
Makes sense, thanks for the context!